tracker/tracker.go

```diff
@@ -24,40 +24,51 @@ func NewStepTracker(logger log.Logger, envRepo env.Repository) StepTracker {
 }
 
 func (t *StepTracker) LogSDKVersions(projectSDKVersions flutterproject.FlutterAndDartSDKVersions) {
+	p := projectSDKVersionsToProperties(projectSDKVersions)
+	t.tracker.Enqueue("flutter_scanner_project_sdk_versions", p)
+}
+
+func (t *StepTracker) Wait() {
+	t.tracker.Wait()
+}
+
+func projectSDKVersionsToProperties(projectSDKVersions flutterproject.FlutterAndDartSDKVersions) analytics.Properties {
 	p := analytics.Properties{}
-	for _, flutterSDK := range projectSDKVersions.FlutterSDKVersions {
-		key := "flutter_sdk_" + string(flutterSDK.Source)
-		value := ""
-
-		if flutterSDK.Version != nil {
-			value = flutterSDK.Version.String()
-		} else if flutterSDK.Constraint != nil {
-			value = flutterSDK.Constraint.String()
-		}
 
-		if value != "" {
-			p[key] = value
+	if projectSDKVersions.FVMFlutterVersion != nil {
+		p["flutter_sdk_fvm_config_json"] = projectSDKVersions.FVMFlutterVersion.String()
+	}
+	if projectSDKVersions.ASDFFlutterVersion != nil {
+		p["flutter_sdk_tool_versions"] = projectSDKVersions.ASDFFlutterVersion.String()
+	}
+	if projectSDKVersions.PubspecLockFlutterVersion != nil {
+		if projectSDKVersions.PubspecLockFlutterVersion.Version != nil {
+			p["flutter_sdk_pubspec_lock"] = projectSDKVersions.PubspecLockFlutterVersion.Version.String()
+		} else if projectSDKVersions.PubspecLockFlutterVersion.Constraint != nil {
+			p["flutter_sdk_pubspec_lock"] = projectSDKVersions.PubspecLockFlutterVersion.Constraint.String()
 		}
 	}
-
-	for _, dartSDK := range projectSDKVersions.DartSDKVersions {
-		key := "dart_sdk_" + string(dartSDK.Source)
-		value := ""
-
-		if dartSDK.Version != nil {
-			value = dartSDK.Version.String()
-		} else if dartSDK.Constraint != nil {
-			value = dartSDK.Constraint.String()
+	if projectSDKVersions.PubspecFlutterVersion != nil {
+		if projectSDKVersions.PubspecFlutterVersion.Version != nil {
+			p["flutter_sdk_pubspec_yaml"] = projectSDKVersions.PubspecFlutterVersion.Version.String()
+		} else if projectSDKVersions.PubspecFlutterVersion.Constraint != nil {
+			p["flutter_sdk_pubspec_yaml"] = projectSDKVersions.PubspecFlutterVersion.Constraint.String()
 		}
-
-		if value != "" {
-			p[key] = value
+	}
+	if projectSDKVersions.PubspecLockDartVersion != nil {
+		if projectSDKVersions.PubspecLockDartVersion.Version != nil {
+			p["dart_sdk_pubspec_lock"] = projectSDKVersions.PubspecLockDartVersion.Version.String()
+		} else if projectSDKVersions.PubspecLockDartVersion.Constraint != nil {
+			p["dart_sdk_pubspec_lock"] = projectSDKVersions.PubspecLockDartVersion.Constraint.String()
+		}
+	}
```

Comment:
This part is repeating and for me makes this function a bit harder to read. What if we create a function on VersionConstraint` which returns a string:
```
func (s VersionConstraint) String() string {
    if s.Version != nil {
        return s.Version.String()
    } else if s.Constraint != nil {
         return s.Constraint.String()
    }
    return ""
}
```
and then it would look a bit nicer with 
```
if projectSDKVersions.PubspecLockDartVersion != nil {
    if  value := projectSDKVersions.PubspecLockDartVersion.String(); value != "" {
        p["dart_sdk_pubspec_lock"] = value
    } 
}
```

Reply:
Updated in https://github.com/bitrise-steplib/bitrise-step-flutter-installer/pull/32/commits/d7c5eeacd5339b72f3a96a722357a33f75284ea5